src/core/server/saved_objects/service/lib/internal_utils.ts

```diff
@@ -141,3 +142,19 @@ export function rawDocExistsInNamespace(
     namespaces?.includes(ALL_NAMESPACES_STRING);
   return existsInNamespace ?? false;
 }
+
+/**
+ * Check to ensure that a 404 response does not come from Elasticsearch
+ *
+ * WARNING: This is a hack to work around for 404 responses returned from a proxy.
+ * We're aiming to minimise the risk of data loss when consumers act on Not Found errors
+ *
+ * @param response response from elasticsearch client call
+ * @returns boolean 'true' if the status code is 404 and the Elasticsearch product header is missing/unexpected value
+ */
+export const isNotFoundFromUnsupportedServer = (args: {
```

Comment:
I'm in two minds about where this helper should go and considered moving it to the `elasticsearch` service but went with `internal_utils` as we're really only using it within the SO service right now.

Reply:
I [moved the method](https://github.com/elastic/kibana/pull/108749/files#diff-a07a07c19da7455a36db61fff13edc39aa9b24d07456b479264b83a74a65f026R28-R33) since we're both thinking it should belong to the `elasticsearch` service. While ATM it's only used in the SO service, other folks might have to implement something similar.